utils/results.py

```diff
@@ -20,4 +21,28 @@ def write_to_csv(self, file_name: str):
                 for i in range(len(self._values[0])):
                     writer.writerow([row[i] for row in self._values])
             else:
-                writer.writerow(self._values)
\ No newline at end of file
+                writer.writerow(self._values)
+
+
+class JsonWriter:
+    def __init__(self, file_name: Optional[str] = None):
+        self._file_name = file_name
+        self.previous = 0
+        self.metrics_list = []
+
+    def add_metrics(self, epochs: int, metrics: dict):
+        if self.metrics_list:
+            for metric in self.metrics_list:
+                if metric["epochs"] == epochs:
+                    metric["metrics"].update(metrics)
+                    return
+            self.metrics_list.append({"epochs": epochs, "metrics": metrics})
+        else:
+            self.metrics_list.append({"epochs": epochs, "metrics": metrics})
+
+    def flush(self):
+        if self._file_name:
+            with open(self._file_name, "a", encoding="UTF8", newline="") as f:
+                f.write(json.dumps(self.metrics_list, indent=4))
+        else:
+            raise ValueError("No file name specified for json writer")
```

Comment:
I needed some easily parse-able data for the PLP to use to select the best epoch. Maybe it was stubborness in me but I thought json was better for this than the already present Results writer.